### PR TITLE
Simplify InstantRFC3339Module by using StdDateFormat from Jackson.

### PR DESCRIPTION
**Pull request by xiaodong-izettle**
Trying to simplify #262 a bit. 

Ping @martinfridh, @adamvoncorswant and @MrShish, thanks.

### CODE REVIEW

**Comment by xiaodong-izettle** [comment]
I copied the test case changes from #262, but ignored one...

File: izettle-jackson/src/test/java/com/izettle/jackson/module/InstantRFC3339ModuleTest.java [hunk 1]
```diff
@@ -16,13 +17,42 @@ private static ObjectMapper createMapper() {
...
+    @Test
+    public void itShouldInterpretOffsetWithColonsCorrectly() throws Exception {
+        final ObjectMapper mapper = createMapper();
+        final Instant parsedInstant = mapper.readValue("\"2016-08-04T09:42:51.336+02:00\"", Instant.class);
+        assertEquals(Instant.parse("2016-08-04T07:42:51.336Z"), parsedInstant);
+    }
+
+    @Test
+    @Ignore
+    public void itShouldInterpretOffsetWithGeneralTimeZoneCorrectly() throws Exception {
```
Comment (xiaodong-izettle) [inline izettle-jackson/src/test/java/com/izettle/jackson/module/InstantRFC3339ModuleTest.java]: @martinfridh, you sure we need to support this? The format `2016-08-04T09:42:51.336UTC+02:00` looks weird to me, probably just me though. 😄

**Comment by adamvoncorswant** [comment]
Cool. would you mind keeping the old way of *serializing* the instant? Then at least, we have explicit code for how we want 'our' format to look like.

**Comment by xiaodong-izettle** [comment]
Sure thing (actually, I don't know why I changed the serialisation part, probably too late now. 😄), updated and force pushed, @adamvoncorswant.

File: izettle-jackson/src/test/java/com/izettle/jackson/module/InstantRFC3339ModuleTest.java (hunk 1 above)
Comment (xiaodong-izettle) [reply izettle-jackson/src/test/java/com/izettle/jackson/module/InstantRFC3339ModuleTest.java]: Thanks for the explanation, @martinfridh. I'll remove this test case then. 😄

**Comment by xiaodong-izettle** [comment]
I'm done, @MrShish and @martinfridh. Feel free to merge. Thanks. 😄

**Comment by xiaodong-izettle** [comment]
Thanks for merging, 😄

**Comment by martinfridh** [comment]
@xiaodong-izettle , you never "released" these changes to local artifactory? do you want to do it, or should I (I guess that how to do it is described in "Release steps" of README)?

**Pull request merged**